x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java

```diff
@@ -4614,7 +4614,6 @@ public void testSimplifyComparisonArithmeticWithConjunction() {
         doTestSimplifyComparisonArithmetics("12 * (-integer - 5) == -120 AND integer < 6 ", "integer", EQ, 5);
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/108525")
     public void testSimplifyComparisonArithmeticWithDisjunction() {
```

Comment:
Apart from this test, I want to see another test that actually shows the effects of adding this rule in the optimizer.
For example, I tested this query `from employees | where 12 * (-languages - 5) >= -120 OR languages < 5 | keep languages` and I looked at how the rules are applied.

Before this rule was added the simplified change to the plan was like this:

```
Rule rules.SimplifyComparisonsArithmetics applied
Limit[1000[INTEGER]]                                                                                           = Limit[1000[INTEGER]]
\_EsqlProject[[languages{f}#172]]                                                                              = \_EsqlProject[[languages{f}#172]]
  \_Filter[NEG(languages{f}#172) - 5[INTEGER] * 12[INTEGER] >= -120[INTEGER] OR languages{f}#172 < 5[INTEGER]] !   \_Filter[NEG(languages{f}#172) - 5[INTEGER] >= -10[INTEGER] OR languages{f}#172 < 5[INTEGER]]
    \_EsRelation[employees][languages{f}#172, languages.byte{f}#173, languages...]                             =     \_EsRelation[employees][languages{f}#172, languages.byte{f}#173, languages...]

Rule rules.SimplifyComparisonsArithmetics applied
EsqlProject[[languages{f}#172]]                                                                 = EsqlProject[[languages{f}#172]]
\_Limit[1000[INTEGER]]                                                                          = \_Limit[1000[INTEGER]]
  \_Filter[NEG(languages{f}#172) - 5[INTEGER] >= -10[INTEGER] OR languages{f}#172 < 5[INTEGER]] !   \_Filter[NEG(languages{f}#172) >= -5[INTEGER] OR languages{f}#172 < 5[INTEGER]]
    \_EsRelation[employees][languages{f}#172, languages.byte{f}#173, languages...]              =     \_EsRelation[employees][languages{f}#172, languages.byte{f}#173, languages...]

Rule rules.SimplifyComparisonsArithmetics applied
EsqlProject[[languages{f}#172]]                                                    = EsqlProject[[languages{f}#172]]
\_Limit[1000[INTEGER]]                                                             = \_Limit[1000[INTEGER]]
  \_Filter[NEG(languages{f}#172) >= -5[INTEGER] OR languages{f}#172 < 5[INTEGER]]  !   \_Filter[languages{f}#172 <= 5[INTEGER] OR languages{f}#172 < 5[INTEGER]]
    \_EsRelation[employees][languages{f}#172, languages.byte{f}#173, languages...] =     \_EsRelation[employees][languages{f}#172, languages.byte{f}#173, languages...]
```

And, in the end, before all transformations are done, I am seeing the final query that gets pushed to ES:

```
ProjectExec[[languages{f}#17]]                                                       = ProjectExec[[languages{f}#17]]
\_LimitExec[1000[INTEGER]]                                                           ! \_FieldExtractExec[languages{f}#17]<[]>
  \_FilterExec[languages{f}#17 <= 5[INTEGER] OR languages{f}#17 < 5[INTEGER]]        !   \_EsQueryExec[employees], indexMode[standard], query[{"bool":{"should":[{"esql_single_value":{"field":"languages","next":{"range":{"languages":{"lte":5,"boost":1.0}}},"source":"12 * (-languages - 5) >= -120@1:24"}},{"esql_single_value":{"field":"languages","next":{"range":{"languages":{"lt":5,"boost":1.0}}},"source":"languages < 5@1:57"}}],"boost":1.0}}][_doc{f}#21], limit[1000], sort[] estimatedRowSize[null]
    \_EsSourceExec[employees][languages{f}#17, languages.byte{f}#18, languages.lo..] !
```

And you see there two `range` queries (in `EsQueryExec`).

After this rule is being added to the optimizer, the simplified chain of rules is like this:


```
Rule rules.SimplifyComparisonsArithmetics applied
Limit[1000[INTEGER]]                                                                                           = Limit[1000[INTEGER]]
\_EsqlProject[[languages{f}#172]]                                                                              = \_EsqlProject[[languages{f}#172]]
  \_Filter[NEG(languages{f}#172) - 5[INTEGER] * 12[INTEGER] >= -120[INTEGER] OR languages{f}#172 < 5[INTEGER]] !   \_Filter[NEG(languages{f}#172) - 5[INTEGER] >= -10[INTEGER] OR languages{f}#172 < 5[INTEGER]]
    \_EsRelation[employees][languages{f}#172, languages.byte{f}#173, languages...]                             =     \_EsRelation[employees][languages{f}#172, languages.byte{f}#173, languages...]

Rule rules.SimplifyComparisonsArithmetics applied
EsqlProject[[languages{f}#172]]                                                                 = EsqlProject[[languages{f}#172]]
\_Limit[1000[INTEGER]]                                                                          = \_Limit[1000[INTEGER]]
  \_Filter[NEG(languages{f}#172) - 5[INTEGER] >= -10[INTEGER] OR languages{f}#172 < 5[INTEGER]] !   \_Filter[NEG(languages{f}#172) >= -5[INTEGER] OR languages{f}#172 < 5[INTEGER]]
    \_EsRelation[employees][languages{f}#172, languages.byte{f}#173, languages...]              =     \_EsRelation[employees][languages{f}#172, languages.byte{f}#173, languages...]

Rule rules.SimplifyComparisonsArithmetics applied
EsqlProject[[languages{f}#172]]                                                    = EsqlProject[[languages{f}#172]]
\_Limit[1000[INTEGER]]                                                             = \_Limit[1000[INTEGER]]
  \_Filter[NEG(languages{f}#172) >= -5[INTEGER] OR languages{f}#172 < 5[INTEGER]]  !   \_Filter[languages{f}#172 <= 5[INTEGER] OR languages{f}#172 < 5[INTEGER]]
    \_EsRelation[employees][languages{f}#172, languages.byte{f}#173, languages...] =     \_EsRelation[employees][languages{f}#172, languages.byte{f}#173, languages...]

Rule rules.CombineBinaryComparisons applied
EsqlProject[[languages{f}#172]]                                                    = EsqlProject[[languages{f}#172]]
\_Limit[1000[INTEGER]]                                                             = \_Limit[1000[INTEGER]]
  \_Filter[languages{f}#172 <= 5[INTEGER] OR languages{f}#172 < 5[INTEGER]]        !   \_Filter[languages{f}#172 <= 5[INTEGER]]
    \_EsRelation[employees][languages{f}#172, languages.byte{f}#173, languages...] =     \_EsRelation[employees][languages{f}#172, languages.byte{f}#173, languages...]
```

So, it is clearly visible in the `EsRelation` that the addition of CombineBinaryComparisons made the actual query that gets pushed to ES be a simpler one (`_Filter[languages{f}#172 <= 5[INTEGER]` instead of `_Filter[languages{f}#172 <= 5[INTEGER] OR languages{f}#172 < 5[INTEGER]`).
I think such a test (that I am assuming we don't have, sorry I haven't checked) would be valuable.

Reply:
Yeah, this is exactly what this rule does, after it is applied, there is only one predicate(`EsqlBinaryComparison`) in the Filter in stead of two(`EsqlBinaryComparison` connected by Or) on languages. 

And the test verifies it as you want it to be :) `doTestSimplifyComparisonArithmetics` calls `extractPlannedBinaryComparison` to verify the top level filter has a `EsqlBinaryComparison` directly under it. 

```
    private void doTestSimplifyComparisonArithmetics(
        String expression,
        String fieldName,
        EsqlBinaryComparison.BinaryComparisonOperation opType,
        Object bound
    ) {
        EsqlBinaryComparison bc = extractPlannedBinaryComparison(expression);
        ...
    }
```
```
    private static EsqlBinaryComparison extractPlannedBinaryComparison(LogicalPlan plan) {
        assertTrue("Expected unary plan, found [" + plan + "]", plan instanceof UnaryPlan);
        UnaryPlan unaryPlan = (UnaryPlan) plan;
        assertTrue("Epxected top level Filter, foung [" + unaryPlan.child().toString() + "]", unaryPlan.child() instanceof Filter);
        Filter filter = (Filter) unaryPlan.child();
        assertTrue(
            "Expected filter condition to be a binary comparison but found [" + filter.condition() + "]",
            filter.condition() instanceof EsqlBinaryComparison
        );
        return (EsqlBinaryComparison) filter.condition();
    }
```

Reply:
Yep, you are right. The test I had in mind does basically the same thing.